store/populations.go

```diff
@@ -97,19 +97,58 @@ func (s *store) GetPlaceObs(ctx context.Context, in *pb.GetPlaceObsRequest,
 			key += "^" + p + "^" + v
 		})
 	}
-	btPrefix := fmt.Sprintf("%s%s", util.BtPlaceObsPrefix, key)
+	key = fmt.Sprintf("%s%s", util.BtPlaceObsPrefix, key)
 
-	// Query for the prefix.
-	btRow, err := s.btTable.ReadRow(ctx, btPrefix)
+	// TODO(boxu): abstract out the common logic for handling cache merging.
+	var baseData, sideData pb.PopObsCollection
+	var baseString, sideString string
+	var hasBaseData, hasSideData bool
+	out.Payload, _ = util.ZipAndEncode("{}")
+
+	btRow, err := s.btTable.ReadRow(ctx, key)
 	if err != nil {
-		return err
+		log.Print(err)
 	}
-	if len(btRow[util.BtFamily]) > 0 {
-		out.Payload = string(btRow[util.BtFamily][0].Value)
+
+	hasBaseData = len(btRow[util.BtFamily]) > 0
+	if hasBaseData {
+		baseString = string(btRow[util.BtFamily][0].Value)
+	}
+	sideString, hasSideData = s.cacheData[key]
+
+	if !hasBaseData && !hasSideData {
+		return nil
+	} else if !hasBaseData {
+		out.Payload = sideString
+		return nil
+	} else if !hasSideData {
+		out.Payload = baseString
+		return nil
 	} else {
-		out.Payload, _ = util.ZipAndEncode("{}")
+		if tmp, err := util.UnzipAndDecode(baseString); err == nil {
+			jsonpb.UnmarshalString(string(tmp), &baseData)
+		}
+		if tmp, err := util.UnzipAndDecode(sideString); err == nil {
+			jsonpb.UnmarshalString(string(tmp), &sideData)
+		}
+		dataMap := map[string]*pb.PopObsPlace{}
+		for _, data := range baseData.Places {
```

Comment:
For a given place, do we for now don't merge the observations? e.g., delta cache has 2018, but base has 2012 to 2017, will it work?

Reply:
place obs has "observationDate" is argument, so no need to merge by years.

In general the "branch cache" has the full dataset, so we can almost always overwrite the base cache, in time, stats_var dimensions.